### PR TITLE
Add new input params

### PR DESCRIPTION
**Pull request by OliviaLynn**
## Problem & Solution Description (including issue #)
Adding in the new input params to follow #35 

## Code Quality
- [x] My code follows [the code style of this project](https://rail-hub.readthedocs.io/en/latest/source/contributing.html#naming-conventions)
- [x] I have written unit tests or justified all instances of `#pragma: no cover`; in the case of a bugfix, a new test that breaks as a result of the bug has been added
- [x] My code contains relevant comments and necessary documentation for future maintainers; the change is reflected in applicable demos/tutorials (with output cleared!) and added/updated docstrings use the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)
- [x] Any breaking changes, described above, are accompanied by backwards compatibility and deprecation warnings

### CODE REVIEW

File: src/rail/estimation/algos/deepdisc.py [hunk 1]
```diff
@@ -39,21 +39,21 @@ def train(config, all_metadata, train_head=True):
     numclasses = config["numclasses"]
     output_dir = config["output_dir"]
     output_name = config["output_name"]
-    period = config["period"]
-    epoch = config["epoch"]
+    epochs_per_print = config["epochs_per_print"]
+    epochs = config["epochs"]
```
Comment (OliviaLynn) [inline src/rail/estimation/algos/deepdisc.py:43]: _(Changed this not to nit, just to match pre-existing stage param)_

File: src/rail/estimation/algos/deepdisc.py [hunk 2]
```diff
@@ -168,6 +167,13 @@ class DeepDiscInformer(CatInformer):
         output_name=Param(str, "deepdisc_informer", required=False, msg="What to call the generated output."),
         chunk_size=Param(int, 100, required=False, msg="Chunk size used within detectron2 code."),
         training_percent=Param(float, 0.8, required=False, msg="The fraction of input data used to split into training/evaluation sets"),
+        num_camera_filters=Param(int, 6, required=False, msg="The number of camera filters for the dataset used (LSST has 6)."),
+        epochs_per_print=Param(int, 5, required=False, msg="How often to print in-progress output."),
+        head_iters=Param(int, 0, required=False, msg="How many iterations in a head run."),
```
Comment (OliviaLynn) [inline src/rail/estimation/algos/deepdisc.py:172]: My grasp of heads and backbones is not quite strong enough to write a meaningful description here - I'm all ears for something better for this

**Review by drewoldag — APPROVED** [review]
Mechanically this looks sound, but I also don't have enough context of `head_iters` and `full_iters`, etc. to write a meaningful description.

File: src/rail/estimation/algos/deepdisc.py (hunk 2 above)
Comment (OliviaLynn) [reply src/rail/estimation/algos/deepdisc.py:172]: head_iters: used when training with a frozen backbone (then, we go and unfreeze the backbone, and run everything)

Comment (OliviaLynn) [inline src/rail/estimation/algos/deepdisc.py:172]:
```suggestion
        head_iters=Param(int, 0, required=False, msg="How many iterations when training the head layers (while the backbone layers are frozen)."),
```

File: src/rail/estimation/algos/deepdisc.py [hunk 3]
```diff
@@ -168,6 +167,13 @@ class DeepDiscInformer(CatInformer):
...
+        full_iters=Param(int, 0, required=False, msg="How many iterations in a full run."),
```
Comment (OliviaLynn) [inline src/rail/estimation/algos/deepdisc.py:173]:
```suggestion
        full_iters=Param(int, 0, required=False, msg="How many iterations when training the head layers and unfrozen backbone layers together."),
```

**Pull request merged**